Classes/CRON/FormBuilder/Controller/FormBuilderController.php

```diff
@@ -88,6 +95,37 @@ public function submitPendingAction()
         $this->view->assign('responseElements', $this->request->getInternalArgument('__responseElements'));
     }
 
+    /**
+     * Checks and decodes the transferred timestamp
+     * @param string $encryptedTime
+     * @return bool
+     * @throws Exception
+     */
+    protected function checkTimestamp($encryptedTime) {
+
+        $minTime = $this->conf['Protection']['minTime'];
+        $maxTime = $this->conf['Protection']['maxTime'];
+
+        $key = $this->conf['Protection']['key'];
+        $cipher = $this->conf['Protection']['cipher'];
+        $iv = $this->conf['Protection']['iv'];
```

Comment:
$key, $iv, that is very hackisch and too low lever to configure here. Maybe use the same key for both, isn't that possible? Or instead of encrypting, use the `Neos\Flow\Security\Cryptography\HashService` class (comes with Flow) and simply create a hash for the time (submit the string "<hash>:<time>") and check if it is still valid (not encrypted, but validatable).

Reply:
Also, this is out as you had already written I use now the function of Flow. :)